src/services/auth.js

```diff
@@ -20,18 +26,119 @@ export function close() {
     jwtKey = undefined;
 }
 
+/**
+ * @param {Buffer|string} data - The data to encode
+ * @returns {string} - base64 encoded string
+ */
+export function base64Encode(data) {
+    if (typeof data === "string") {
+        data = Buffer.from(data);
+    }
+    return data.toString("base64");
+}
+
+/**
+ * @param {string} str base64 encoded string
+ * @returns {Buffer}
+ */
+function base64Decode(str) {
+    let output = str;
+    const paddingLength = 4 - (output.length % 4);
+    if (paddingLength < 4) {
+        output += "=".repeat(paddingLength);
+    }
+    return Buffer.from(output, "base64");
+}
+
+/**
+ * Signs and creates a JWT token
+ *
+ * @param {Object} payload - The payload to include in the token
+ * @param {WithImplicitCoercion<string>} [key] - Optional key, defaults to the configured jwtKey
+ * @param {Object} [options]
+ * @param {string} [options.algorithm] - The algorithm to use, defaults to HS256
+ * @returns {string} - The signed JWT token
+ * @throws {AuthenticationError}
+ */
+export function sign(payload, key = jwtKey, { algorithm = ALGORITHM.HS256 } = {}) {
```

Comment:
Do we need to bother with all of these params if we're only using the default one?

In particular if `ALGORITHM_FUNCTIONS` doesn't support any other one anyway?

Reply:
The code is heavily inspired on the lib, IMO this is fine to copy them even if we only have a single algorithm at hand now.

Reply:
I kept them to keep the expected API of JWT functions and to make it more clear in the code what are JWT core features vs parameters, it's also more similar to our python implementation that way